src/Language/PureScript/Sugar/TypeClasses/Deriving.hs

```diff
@@ -72,8 +71,12 @@ deriveInstances
   -> Module
   -> m Module
 deriveInstances externs (Module ss coms mn ds exts) =
-    Module ss coms mn <$> mapM (deriveInstance mn synonyms instanceData ds) ds <*> pure exts
+    Module ss coms mn <$> mapM (deriveInstance mn synonyms kinds instanceData ds) ds <*> pure exts
   where
+    -- TODO: Do we need to reconstruct kind information?
```

Comment:
Were you intending to address this in this PR?

Reply:
No, I'll remove the comment. This is the same issue as typeclass desugaring happening too soon. These passes happen before typechecking, but they need to do type synonym expansion. If you try to use a polykinded synonym here that reifies a kind (like `KindOf`) then you'll get an incomplete, unhygienic expansion I think.

Reply:
Deriving should be done on demand as part of instance checking.

Reply:
Makes sense. Do you think it’s worth making a separate issue for this then?

Reply:
Yes, for sure.